source/common/network/address_impl.cc

```diff
@@ -233,6 +237,14 @@ std::string Ipv6Instance::Ipv6Helper::makeFriendlyAddress() const {
   ASSERT(str == ptr);
   return ptr;
 }
+StatusOr<InstanceConstSharedPtr> Ipv6Instance::Ipv6Helper::v4CompatibleAddress() const {
+  if (!v6only_ && IN6_IS_ADDR_V4MAPPED(&address_.sin6_addr)) {
+    struct sockaddr_in sin;
+    ipv6ToIpv4CompatibleAddress(&address_, &sin);
+    return Address::InstanceFactory::createInstancePtr<Address::Ipv4Instance>(&sin);
+  }
+  return absl::InvalidArgumentError("It isn't Ipv4-mapped Ipv6 address");
```

Comment:
Instead of using StatusOr perhaps just return nullptr? I'm not sure the status provides much value here?

Reply:
yes, agree with you, will update